### ISSUE TITLE
Play project when hitting space

### ISSUE THREAD

**Issue by iamdannywyatt** [DISTRHO/Cardinal#512]
In Logic, at least, if Cardinal is upfront, when I hit Space, Logic doesn't start, which is a bit annoying, because I then have to click outside Cardinal to make it start. It's just extra clicks that could be avoided if SPACE would automatically play the project like any other plugin

**Comment by dromer** [comment]
But how would you then be able to type a space in any of the text input boxes?
In general it is unwanted that the DAW responds to such keys when focus is put on the plugin. It would be mighty annoying if the DAW starts playing/stopping/playing when you are just typing some text into the plugin.

**Comment by iamdannywyatt** [comment]
> But how would you then be able to type a space in any of the text input boxes?
> In general it is unwanted that the DAW responds to such keys when focus is put on the plugin. It would be mighty annoying if the DAW starts playing/stopping/playing when you are just typing some text into the plugin.

The plugin would know when text fields were active. I just tried with both stock plugins and Valhalla SuperMassive on fields where I can double click to input a number and using space didn't start the project.

**Comment by falkTX** [comment]
This could in theory work correctly in VST, CLAP or LV2, but cant work via AU because of the way it is set up at the moment.

**Comment by iamdannywyatt** [comment]
I'm sure that's a more technical aspect that I wouldn't be able to understand, since I'm just a user and coding isn't my thing. When you say "at the moment" you mean it could potentially be implemented at all?

**Comment by falkTX** [comment]
Correct. Right now the AU is a hack on getting JUCE to do the heavy lifting for us, and we just place the Cardinal UI on top of the JUCE UI (that is empty).
The proper way to go about it is to finish the implementation of AU directly on the plugin framework in use by Cardinal. But interest is very low compared to other formats as it is macOS specific, and besides GarageBand and Logic Pro, pretty much all DAWs support at least one of the other formats.

**Comment by dromer** [comment]
@alltiagocom have you tried the latest AU builds?
